tests/functional/demo/test_demo.py

```diff
@@ -157,3 +159,55 @@ def test_workon(database):
     assert params[0]['name'] == '/x'
     assert params[0]['type'] == 'real'
     assert (params[0]['value'] - 34.56789) < 1e-5
+
+
+@pytest.mark.usefixtures("clean_db")
+def test_stress_unique_folder_creation(database, monkeypatch, tmpdir, capfd):
+    """Test integration with a possible framework that needs to create
+    unique directories per trial.
+    """
+    # XXX: return and complete test when there is a way to control random
+    # seed of Oríon
+    how_many = 1000
+    monkeypatch.chdir(os.path.dirname(os.path.abspath(__file__)))
+    process = subprocess.Popen(["orion", "--max-trials={}".format(how_many),
+                                "--pool-size=1",
+                                "--name=lalala",
+                                "--config", "./stress_gradient.yaml",
+                                "./dir_per_trial.py",
+                                "--dir={}".format(str(tmpdir)),
+                                "--other-name~exp.name",
+                                "--name~trial.hash_name",
+                                "-x~gaussian(30, 10)"])
+    rcode = process.wait()
+
+    exp = list(database.experiments.find({'name': 'lalala'}))
+    assert len(exp) == 1
+    exp = exp[0]
+    assert '_id' in exp
+    exp_id = exp['_id']
+
+    assert rcode == 0
+    # For contingent broken trials, which in this test means that a existing
+    # directory was attempted to be created, it means that it's not md5 or
+    # bad hash creation to blame, but the finite precision of the floating
+    # point representation. Specifically, it seems that gradient descent
+    # is able to reach such levels of precision jumping around the minimum
+    # (notice that an appropriate learning rate was selected in this stress
+    # test to create underdamped behaviour), that it begins to suggest same
+    # things from the past. This is intended to be shown with the assertions
+    # in the for-loop below.
```

Comment:
Why not used random search if gradient descent causes side-effects you need to verify to make sure the problem is not md5 of bad hash creation.

Reply:
For the reason that even in the case where `Δx_t` is the minimum possible in the finite float representation scenario, which can be reached by gradient descent for example, the demand for `diff hash_names => diff parameters` is true. By random search I wouldn't be able to have that stressed behaviour.